[FEAT] Bumps to web3 v6, adds linters, adds tests (pending upstream merge)

8ball030 opened the pull request:
This pr is to (Hopefully) bump balpy to web3.py > v5

There are a number of dependants which would need to be merged and released;

https://github.com/gerrrg/multicaller/pull/1

I am very sorry about the size of this pr i got a little carried away.

However, it does accomplish few things and fixes a few bugs caught by the now introduced linters.

Features.

-> formatters to fix the incostent tabs and spaces issues, sort imports and format with black.

-> linters: -> autopep8

-> test suite: demonstrates functionality of the subgraph queries to allow for tests.

-> dev group for dependencies.


Probably a bit too aggressive i again apologies.

8ball030 committed:
bumped to web6

Commit:
[feat] deps

8ball030 committed:
bumped to web3v6

8ball030 committed:
small improvements

8ball030 committed:
reverted testing changes

8ball030 committed:
added test for graph

8ball030 committed:
[chore] ensured pep8 compliance

8ball030 committed:
[chore] ensured pep8 compliance

8ball030 committed:
[chore] ensured pep8 compliance

8ball030 committed:
[chore] added dev deps group

8ball030 committed:
[chore] added docs

8ball030 commented on makefile:
```diff
@@ -0,0 +1,13 @@
+test:
+	poetry run pytest tests
+
+all: fmt lint test
```
consistent formatting and linting

8ball030 committed:
added test for graph

8ball030 commented on tests/test_graph.py:
```diff
@@ -0,0 +1,12 @@
+TEST_NETWORK = "gnosis-chain"
+
+
+def test_get_pool_ids():
+    get_pool_ids(TEST_NETWORK)
+
+
+def test_get_pools():
+    get_pools(TEST_NETWORK)
```
tests for the graph queries.

gerrrg commented on pyproject.toml:
```diff
@@ -6,19 +6,26 @@ authors = ["Greg <greg@orb.cl>"]
+gql=">=2.0.0"
+jstyleson=">=0.0.2"
+web3="~6"
+multicaller = {path = "../multicaller"}
```
todo use version number

gerrrg commented on balpy/balancerv2cad/src/balancerv2cad/WeightedPool.py:
There's some history around this -- it used to be a standalone package, was briefly a submodule, then was directly thrown into the package when the previous maintainer said they had no plans to make any future updates. It seemed safer to just move it into balpy to have the freedom to update as needed.

That being said, I am not aware of anyone using it, and the packages do have some errors in them. I've had someone find some notable issues in the stable math library, and the way that the "pools" themselves are structured is incorrect.

At this point, it might just make sense to remove them entirely, at least for now. It probably doesn't sense to move these files forward knowing that they have accuracy/implementation issues

8ball030 committed:
[feat] ensured commit matches

8ball030 committed:
Merge pull request #1 from 8ball030/feat/syncing-changes-2

[feat] ensured commit matches

8ball030 committed:
feat: added base

8ball030 committed:
feat: added base

8ball030 committed:
feat: add apiEndpoint to balpy class

8ball030 committed:
feat: relaxed dependencies

8ball030 committed:
Merge pull request #3 from 8ball030/feat/improvments

feat: relaxed dependencies

8ball030 committed:
Merge pull request #4 from 8ball030/feat/stable

feat: add apiEndpoint to balpy class

8ball030 committed:
feat: relaxed dependencies

Commit:
feat: updated for v3 api

Commit:
chore: relaxed deps

8ball030 committed:
Merge branch 'main' into feat/updated-upstream

Commit:
chore: regenerated poetry lock

8ball030 committed:
Merge pull request #5 from 8ball030/feat/updated-upstream

chore: relaxed deps

Commit:
feat:fixed import

Commit:
chore: linters and api3 integration

Commit:
fix: retry to account for blocking

8ball030 committed:
Merge branch 'main' into feat/updates

8ball030 committed:
Merge pull request #6 from 8ball030/feat/updates

chore: linters and api3 integration

Commit:
chore: removed unused code

Commit:
chore: improve-balpy api response handling to advoid rpc

Commit:
chore: improve copy of query

Commit:
chore:workflows-and-github

Commit:
feat/ensured-passes-ci

8ball030 committed:
Merge pull request #7 from 8ball030/feat/preparing-fork

chore:workflows-and-github

Commit:
feat:tbump

Commit:
Bump to 0.0.2

Commit:
Bump to 0.0.3

Commit:
feat:bumping

Commit:
Bump to 0.0.4

Commit:
feat:bumping

Commit:
Bump to 0.0.5

Commit:
feat:added-common-checks

Commit:
Bump to 0.0.6

Commit:
feat:aligning-relase-versions

Commit:
Bump to 0.0.7

Commit:
feat:refactor

8ball030 committed:
Merge pull request #8 from 8ball030/feat/extra-endpoints

feat:refactor

Commit:
Bump to 0.0.8

Commit:
Bump to 0.0.9

gauravlochab committed:
feat:Add MODE support in balpy

gauravlochab committed:
chore:version update to 0.1.0 for release

gauravlochab committed:
feat: Add mode.json for various tasks
This commit adds mode.json files for multiple tasks in the balpy/balancer-deployments directory.

gauravlochab committed:
fix: resolve ci issue for lockfile

8ball030 committed:
Merge pull request #9 from 8ball030/feat/add-mode

Feat/add mode

Commit:
fix:added-deployments-docs-table

Commit:
Bump to 0.1.1

Commit:
Bump to 0.1.1

xiuxiuxar committed:
feat: return batch swap call data

xiuxiuxar committed:
set SOR to find v2 swaps only and return only bytes32 poolids

xiuxiuxar committed:
revert to correct poolid logic, now that API returns v2 only

xiuxiuxar committed:
define sender and recipient for safe address compatibility

xiuxiuxar committed:
remove unneeded code

xiuxiuxar committed:
fix ci, bump versions

xiuxiuxar committed:
bump ci poetry

xiuxiuxar committed:
update lockfile

xiuxiuxar committed:
bump ci poetry

Commit:
feat:fixed-ci-issues

Commit:
feat:fixed-ci-issues

Commit:
feat:fixed-ci-issues

Commit:
fix:lock-issues

8ball030 committed:
Merge branch 'main' into test-fixes

Commit:
fix:lock-issues

Commit:
fix:linter-issues

8ball030 committed:
Merge pull request #13 from 8ball030/test-fixes

feat:fixed-ci-issues

8ball030 committed:
Merge pull request #10 from 8ball030/feat/generate-deployments-docs-table

fix:added-deployments-docs-table

8ball030 committed:
Merge branch 'fix/swaps' into chore/fix-swap-data

8ball030 committed:
Merge pull request #12 from xiuxiuxar/chore/fix-swap-data

Chore/fix swap data

Commit:
fix:relock-poetry

Commit:
chore:linters

8ball030 committed:
Merge pull request #14 from 8ball030/fix/swaps

Fix/swaps

xiuxiuxar committed:
feat: add get ticker graph query

8ball030 committed:
Merge pull request #15 from xiuxiuxar/feat/add-ticker-query

feat: add get ticker graph query

Commit:
Bump to 0.1.2

Commit:
feat:example-and-improvements

Commit:
chore:linters

Commit:
Bump to 0.1.3

Commit:
feat:fixed-query-to-pin-protocol

Commit:
Bump to 0.1.4

xiuxiuxar committed:
bump multicaller -> v0.3.0

xiuxiuxar committed:
Merge branch 'main' into chore/bump-multicaller

8ball030 committed:
Merge pull request #17 from xiuxiuxar/chore/bump-multicaller

Chore/bump multicaller

8ball030 committed:
Merge pull request #18 from 8ball030/bump-multicaller

Bump multicaller

Commit:
Bump to 0.1.5

xiuxiuxar committed:
remove web3 middleware

8ball030 committed:
Merge pull request #19 from xiuxiuxar/chore/remove-web3-middleware

remove web3 middleware

Commit:
Bump to 0.1.6

Commit:
feat:fixing-deps
